### ISSUE TITLE
NVDA screen reader not working in OcApi and some extra information about this.

### ISSUE THREAD

**Issue by Sergiaws** [shorthorn-project/One-Core-API-Source#80]
NVDA screen reader used to work quite well on vanilla XP, but in OcAPI doesn't work. There was a Skype group called Windows XP Forever where this was mentioned, but an other person told that NVDA don't work in OcAPI because of the usage of Wine dependencies. As I found that is totally false. There is a [Script for installing NVDA and ESpeak on Wine for Mac and Linux](https://nashcentral.duckdns.org/projects/wineinstall) and it works even better than NVDA in XP with OcAPI!

The problem is that on XP+OcAPI NVDA just reads the title of the Window, not other content. In Wine it can be buggy in some circunstances, but is seriously more usable than in XP+OcAPI! You could see that if you install Wine, install extra dependencies (need Lynx and cabextractor) and run that script. Then run NVDA and when you see it is speaking try to open some app (test winver and move with arrows or tab and you could see that it works. You can also try with "wine uninstall", classic applet for uninstalling programs on Windows, but for Wine. It works also well with NVDA

Another thing I found is that if you try to install One Core API with NVDA running, it will throw an error regarding Oleacc.dll and I found that it's related with UI and accessibility, and it's look like OcAPI setup aborts because of the inhability of upgrading that dll. I also found that Windows Vista Extended Kernel made by [@Win32](https://github.com/win32ss) don't have such issues, and when @Win32 started more seriously to add 32 bit functions to WinVista exKernel I was able to upgrade NVDA 2017.3 (latest version for XP/Vista) to 2023.x (one of the latest versions for Win7). NVDA also works quite well in Win2K even with Blackwingcat's extended kernel. You considered to port Aero to XP, but Aero seems to be just a Theme, it might not affect NVDA at all. NVDA can be affected positively if you could backport Win7+ UIA to XP,and no idea if WinVista/7+ versions of dlls like Oleacc.dll could be ported to XP.

Some extra things that might need to be done.
- Compare opening desktop and browsing with arrows in vanilla XP and XP+OcAPI,you can also test that writing some lines in Notepad and open menu with alt and browsing with arrows.
- Try to analize NVDA with dependency walker or similar stuff to see if there are other dlls involved.
- Try to test Wine script I posted here in Linux if you consider it relevant.

**Comment by jonm58** [comment]
<img width="1598" height="754" alt="Image" src="https://github.com/user-attachments/assets/5a3ab5fa-ea64-40ea-b6ab-0fc36f958fdf" />
Your issue should be posted to the issues in One-Core-API-Binaries
https://github.com/shorthorn-project/One-Core-API-Binaries/issues
